compiler/src/dotty/tools/dotc/transform/MegaPhase.scala

```diff
@@ -296,9 +296,8 @@ class MegaPhase(val miniPhases: Array[MiniPhase]) extends Phase {
         }
       case tree: Block =>
         inContext(prepBlock(tree, start)(using outerCtx)) {
-          val stats = transformStats(tree.stats, ctx.owner, start)
-          val expr = transformTree(tree.expr, start)
-          goBlock(cpy.Block(tree)(stats, expr), start)
+          val stats1 = transformStats(tree.stats :+ tree.expr, ctx.owner, start)
```

Comment:
This would be terrible for performance!

Reply:
I agree, but this code is not new; it is only being moved up from `TreeMapWithImplicits` to `TreeMapWithPreciseStatContexts`.

Reply:
I'm confused. That code is in MegaPhase.

Reply:
> I guess it's bad everywhere, right?

TreeMapWithImplicits is probably called two orders of magnitude less than the other two places. So this makes a big difference.

Reply:
The best technique might be to add the expr as additional argument to transformStats

Reply:
Changes in MegaPhase deleted